packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2908,6 +2910,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       // Either way, we're looking at a '<': tt.jsxTagStart or relational.
 
       let typeParameters: ?N.TsTypeParameterDeclaration;
+      let hasSingleTypeError = { status: false, start: 0, identifierName: "" };
```

Comment:
@JLHwung I needed to raise error outside `const arrow = ....` method because it should return `expr` end of it instead an error.
This just a quick type which is defined in here but if you suggest me to move the type to anywhere else, that is okay too.

Reply:
Can we do `let invalidSingleType: ?N.TSTypeParameter`, and then just do `invalidSingleType = parameter` when it's invalid? It's similar to what we already do with the `typeParameters` variable, and we avoid allocationg a few objects unnecessary.